docs/components/reports.rst

```diff
@@ -1,4 +1,220 @@
 Reports
 #######
+To add and render custom reports in Mautic, your plugin needs to listen to three events:
 
+- ``\Mautic\ReportBundle\ReportEvents::REPORT_ON_BUILD``
+- ``ReportEvents::REPORT_ON_GENERATE``
+- ``ReportEvents::REPORT_ON_GRAPH_GENERATE``
+
+This guide walks you through defining a custom report, generating report data, and rendering graphs.
+
+Defining the Report
+-------------------
+
+Use the ``ReportEvents::REPORT_ON_BUILD`` event to define:
+
+- The report context
+- Available columns
+- Available filters (defaults to columns)
+- Available graphs
+
+Column Definition
+-----------------
+
+Each column array can include the following properties:
+
+.. vale on
+
+.. list-table::
+    :header-rows: 1
+
+    * - Key
+      - Required?
+      - Type
+      - Description
+    * - ``label``
+      - REQUIRED
+      - string
+      - The language string for the column.
+    * - type
+      - REQUIRED
+      - string
+      - Column type.
+    * - alias
+      - OPTIONAL
+      - string
+      - An alias for the returned value. Useful in conjuction with ``formula``.
```

Comment:
```suggestion
      - An alias for the returned value. Useful in conjunction with ``formula``.
```

Vale picked up the typo - @OfficialJhimmy please use Vale within your editor to save reviewers having to snag these issues. Shout if you need help with setting that up, VSCode should work out of the box with Vale.